tools/slang-test/render-api-util.cpp

```diff
@@ -0,0 +1,210 @@
+};
+
+static int _calcAvailableApis()
+{
+    int flags = 0;
+    for (int i = 0; i < int(RenderApiType::kCountOf); i++)
+    {
+        if (RenderApiUtil::calcHasApi(RenderApiType(i)))
+        {
+            flags |= (1 << i);
+        }
+    }
+
+    return flags;
+}
+
+/* static */int RenderApiUtil::getAvailableApis()
+{
+    static int s_availableApis = _calcAvailableApis();
+    return s_availableApis;
+}
+
+/* static */RenderApiType RenderApiUtil::findApiTypeByName(const Slang::UnownedStringSlice& name)
+{
+    using namespace Slang;
+    List<UnownedStringSlice> namesList;
+    for (int j = 0; j < SLANG_COUNT_OF(RenderApiUtil::s_infos); j++)
+    {
+        const auto& apiInfo = RenderApiUtil::s_infos[j];
+        const UnownedStringSlice names(apiInfo.names);
+
+        if (names.indexOf(',') >= 0)
+        {
+            StringUtil::split(names, ',', namesList);
+            if (namesList.IndexOf(name) != UInt(-1))
+            {
+                return apiInfo.type;
+            }
+        }
+        else if (names == name)
+        {
+            return apiInfo.type;
+        }
+    }
+    return RenderApiType::kUnknown;
+}
+
+/* static */int RenderApiUtil::findApiFlagsByName(const Slang::UnownedStringSlice& name)
+{
+    // Special case 'all'
+    if (name == "all")
+    {
+        return int(RenderApiFlag::kAllOf);
+    }
+    RenderApiType type = findApiTypeByName(name);
+    return (type == RenderApiType::kUnknown) ? 0 : (1 << int(type));
+}
+
+/* static */Slang::Result RenderApiUtil::parseApiFlags(const Slang::UnownedStringSlice& text, int* apiBitsOut)
+{
+    using namespace Slang;
+
+    int apiBits = 0;
+
+    List<UnownedStringSlice> slices;
+    StringUtil::split(text, ',', slices);
+
+    for (int i = 0; i < int(slices.Count()); ++i)
+    {
+        UnownedStringSlice slice = slices[i];
+        bool add = true;
+        if (slice.size() <= 0)
+        {
+            return SLANG_FAIL;
+        }
+        if (slice[0] == '+')
+        {
+            // Drop the +
+            slice = UnownedStringSlice(slice.begin() + 1, slice.end());
+        }
+        else if (slice[0] == '-')
+        {
+            add = false;
+            // Drop the +
+            slice = UnownedStringSlice(slice.begin() + 1, slice.end());
+        }
+
+        // We need to find the bits... 
+        int bits = findApiFlagsByName(slice);
+        // 0 means an error
+        if (bits == 0)
+        {
+            return SLANG_FAIL;
+        }
+
+        if (add)
+        {
+            apiBits |= bits;
+        }
+        else
+        {
+            apiBits &= ~bits;
+        }
+    }
+
+    *apiBitsOut = apiBits;
+    return SLANG_OK;
+}
+
+/* !!!!!!!!!!!!!!!!!!!!!!!!!!!! Platform specific stuff !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! */
+
+
+#if SLANG_WINDOWS_FAMILY
+
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+#undef WIN32_LEAN_AND_MEAN
+#undef NOMINMAX
+
+namespace { // anonymous
+
+class WinModule
+{
```

Comment:
I can't remember, but I think there is some code in `core` or `slang-test` for loading dynamic libraries in a platform-independent fashion. Maybe this logic could use that.

Yeah, I just looked and `core` defines a `SharedLibrary` type that might be usable for this.

Not worth blocking the PR, but we may want to reduce the overlap later (and maybe redirect some of the existing `LoadLibrary` calls in slang over to whatever utility class we keep).

Reply:
ok, makes sense - i'll check it out